flow360/component/volume_mesh.py

```diff
@@ -772,3 +786,192 @@ def __getitem__(self, index) -> VolumeMesh:
     # pylint: disable=useless-parent-delegation
     def __iter__(self) -> Iterator[VolumeMesh]:
         return super().__iter__()
+
+
+###==== V2 API version ===###
+
+
+class VolumeMeshStatusV2(Enum):
+    """Status of volume mesh resource, the is_final method is overloaded"""
+
+    SUBMITTED = "submitted"
+    UPLOADING = "uploading"
+    UPLOADED = "uploaded"
+    COMPLETED = "completed"
+
+    def is_final(self):
+        """
+        Checks if status is final for volume mesh resource
+
+        Returns
+        -------
+        bool
+            True if status is final, False otherwise.
+        """
+        if self in [VolumeMeshStatusV2.COMPLETED]:
+            return True
+        return False
+
+
+class VolumeMeshMetaV2(AssetMetaBaseModelV2):
+    """
+    VolumeMeshMetaV2 component
+    """
+
+    project_id: str = pd.Field(alias="projectId")
+    deleted: bool = pd.Field()
+    status: VolumeMeshStatusV2 = pd.Field()  # Overshadowing to ensure correct is_final() method
+
+
+class VolumeMeshDraftV2(ResourceDraft):
+    """
+    Volume mesh draft component
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        file_names: List[str],
+        project_name: str = None,
+        solver_version: str = None,
+        length_unit: LengthUnitType = "m",
+        tags: List[str] = None,
+    ):
+        if not isinstance(file_names, list) or len(file_names) != 1:
+            raise Flow360FileError(
+                "file_names field has to be a single-element list for the volume mesh draft."
+            )
+
+        self.file_name = file_names[0]
+        self.project_name = project_name
+        self.tags = tags if tags is not None else []
+        self.length_unit = length_unit
+        self.solver_version = solver_version
+        self.format = None
+        self._validate()
+        ResourceDraft.__init__(self)
```

Comment:
this could be handled by pydantic model. Leave ticket for potential future improvement

Reply:
Agreed. Left it as-is for now because of how AssetBase from_file method is implemented.